### PR TITLE
ccache: add version 4.8.2

### PR DESCRIPTION
**Pull request by Cogitri**
Also update the dependencies

Specify library name and version:  **ccache/4.8.2**

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by jcar87** [comment]
The build errors appear to be a manifestation of https://github.com/ccache/ccache/issues/1278, for which there is a patch that made it to `4.8.2` - I will backport and reference this.

**Comment by Cogitri** [comment]
Ah grand, thanks

**Comment by jcar87** [comment]
Hi @Cogitri - I did not have permission to push to your branch, so I've opened an independent PR to fix the build issue.

https://github.com/conan-io/conan-center-index/pull/18542

Feel free to cherry-pick the commit from my branch into this PR if you would like this new version and the version bumps to be merged quicker - otherwise once/if my PR is merged first, you can rebase your PR to sync up with the latest changes.

Also thanks for fixing the issue with zstd! very nice catch

File: recipes/ccache/all/conanfile.py [hunk 1]
```diff
@@ -67,7 +67,8 @@ def validate(self):
             raise ConanInvalidConfiguration(f"{self.ref} requires C++ filesystem library, that is not supported by Clang 11 + libstdc++.")
 
     def build_requirements(self):
-        self.tool_requires("cmake/3.25.3")
+        self.tool_requires("cmake/[>=3.15 <4]")
+        self.tool_requires("pkgconf/1.9.3") # for zstd
```
Comment (jcar87) [inline recipes/ccache/all/conanfile.py:71]: This doesn't work properly on Windows from what I can see.

My advice would be to add:
```
deps.set_property("zstd", "cmake_find_mode", "module")
```

further down below (around line 89), which will cause Conan to generate `Findzstd.cmake` which will be used instead, and then pkgconf is no longer needed.

**Review by jcar87 — CHANGES_REQUESTED** [review]
Thanks for your contribution!

In order to fix the build errors on Windows, please add the following in the `generate()` method:
```
deps.set_property("zstd", "cmake_find_mode", "module")
``` 

which should make the tool_requires on pkgconf unnecessary

**Review by jcar87 — APPROVED** [review]
Thanks a lot for your contribution!

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**